packages/sdk/src/Incentives/ERC20PeggedVariableCriteriaIncentive.ts

```diff
@@ -0,0 +1,563 @@
+ *     typeof erc20PeggedVariableCriteriaIncentiveAbi
+ *   >]
+ */
+export type Erc20PeggedVariableCriteriaIncentiveLog<
+  event extends ContractEventName<
+    typeof erc20PeggedVariableCriteriaIncentiveAbi
+  > = ContractEventName<typeof erc20PeggedVariableCriteriaIncentiveAbi>,
+> = GenericLog<typeof erc20PeggedVariableCriteriaIncentiveAbi, event>;
+
+/**
+ * A simple ERC20 incentive implementation that allows claiming of tokens
+ *
+ * @export
+ * @class Erc20PeggedVariableCriteriaIncentive
+ * @typedef {ERC20PeggedVariableCriteriaIncentive}
+ * @extends {DeployableTarget<ERC20PeggedVariableCriteriaIncentive>}
+ */
+export class ERC20PeggedVariableCriteriaIncentive extends DeployableTarget<
+  ERC20PeggedVariableCriteriaIncentivePayload,
+  typeof erc20PeggedVariableCriteriaIncentiveAbi
+> {
+  public override readonly abi = erc20PeggedVariableCriteriaIncentiveAbi;
+  /**
+   * @inheritdoc
+   *
+   * @public
+   * @static
+   * @type {Record<number, Address>}
+   */
+  public static override bases: Record<number, Address> = {
+    31337: import.meta.env.VITE_ERC20_INCENTIVE_BASE,
+    ...(ERC20IncentiveBases as Record<number, Address>),
+  };
+  /**
+   * @inheritdoc
+   *
+   * @public
+   * @static
+   * @type {RegistryType}
+   */
+  public static override registryType: RegistryType = RegistryType.INCENTIVE;
+
+  /**
+   * The owner of the incentive
+   *
+   * @public
+   * @async
+   * @param {?ReadParams} [params]
+   * @returns {Promise<Address>}
+   */
+  public async owner(params?: ReadParams) {
+    return await readErc20PeggedVariableCriteriaIncentiveOwner(this._config, {
+      address: this.assertValidAddress(),
+      args: [],
+      // biome-ignore lint/suspicious/noExplicitAny: Accept any shape of valid wagmi/viem parameters, wagmi does the same thing internally
+      ...(params as any),
+    });
+  }
+
+  /**
+   * Calculates the current reward based on the time since the last claim. The reward is calculated based on the time since the last claim, the available budget, and the reward parameters. It increases linearly over time in the absence of claims, with each hour adding `rewardBoost` to the current reward, up to the available budget. For example, if there is one claim in the first hour, then no claims for three hours, the claimable reward would be `initialReward - rewardDecay + (rewardBoost * 3)`
+   *
+   * @public
+   * @async
+   * @param {?ReadParams} [params]
+   * @returns {Promise<bigint>} - The current reward
+   */
+  public async currentReward(params?: ReadParams) {
+    return await readErc20PeggedVariableCriteriaIncentiveCurrentReward(
+      this._config,
+      {
+        address: this.assertValidAddress(),
+        args: [],
+        // biome-ignore lint/suspicious/noExplicitAny: Accept any shape of valid wagmi/viem parameters, wagmi does the same thing internally
+        ...(params as any),
+      },
+    );
+  }
+
+  /**
+   * Fetches the IncentiveCriteria struct from the contract
```

Comment:
/**
   * Retrieves the maximum reward amount per claim from the contract.
   *
   * @param {?ReadParams} [params] Optional parameters for the read operation
   * @returns {Promise<bigint>} The maximum reward amount per claim
   */